### PR TITLE
feat: prevent timestamp from being read out twice on iOS

### LINKED ISSUE #1449: Each timestamp is read out twice on iOS voiceover
**Describe the bug**
Using voiceover on IOS navigate to the timestamp hear it read out twice.

Could be related to https://github.com/bbc/simorgh/pull/1373/files#diff-0e47b297888cf8b4c4eaef5dace2cfd4R38 (and would therefore be fixed by #1446).

**To Reproduce**

1. On an iOS device, turn on VoiceOver
2. Visit ~~https://www.bbc.com/news/articles/cxvxrj9yvppo~~ (this article id is no longer in use; try https://www.bbc.co.uk/news/articles/c5ll353v7y9o or other articles)
3. Navigate to the timestamp
4. Observe that the timestamp is read out twice

**Expected behavior**
The timestamp should only be read out once.

**Screenshots**
N/A

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
Any iOS device

**Additional context**
N/A

- [x] Initially labelled with ["bug"](https://github.com/bbc/simorgh/labels/bug)

### PR DESCRIPTION
**Pull request by mykeels**
Resolves https://github.com/bbc/simorgh/issues/1449

**Overall change:** _prevent timestamp from being read out twice on iOS._

**Code changes:**

- _wrap &lt;time> with &lt;span role="text"/>._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

**Review by rhenshaw56 — APPROVED** [review]
Looks good. Tested on iphone and timestamp is read out once

**Review by simonsinclair — COMMENTED** [review]
This is a great investigation, @mykeels. I have a couple of reservations about these changes though and most of them are echoed by @tochwill's [comment in the parent issue](https://github.com/bbc/simorgh/issues/1449#issuecomment-544518175).

Ultimately wrapping the `time` element in `role="text"` (which [hasn't been adopted by Aria](https://www.w3.org/TR/wai-aria/#roles_categorization)) changes the semantics of that element. And although the change may _temporarily_ fix this issue in iOS, it doesn't target iOS and will affect all other platforms too. Does this issue present itself as a large enough UX issue to use a hack like this in our code?

Cc @greenc05

**Comment by andrewscfc** [comment]
> This is a great investigation, @mykeels. I have a couple of reservations about these changes though and most of them are echoed by @tochwill's [comment in the parent issue](https://github.com/bbc/simorgh/issues/1449#issuecomment-544518175).
> 
> Ultimately wrapping the `time` element in `role="text"` (which [hasn't been adopted by Aria](https://www.w3.org/TR/wai-aria/#roles_categorization)) changes the semantics of that element. And although the change may _temporarily_ fix this issue in iOS, it doesn't target iOS and will affect all other platforms too. Does this issue present itself as a large enough UX issue to use a hack like this in our code?
> 
> Cc @greenc05

Where the conversation is at at time of writing: https://github.com/bbc/simorgh/issues/1449#issuecomment-573640710

File: packages/containers/psammead-timestamp-container/src/index.test.jsx [hunk 1]
```diff
@@ -95,8 +95,9 @@ describe('Timestamp', () => {
           />,
         );
 
-        const time = container.querySelector('time');
-        expect(time.textContent).toEqual('27 مهر 1397 - ۱۹ اکتبر ۲۰۱۸');
+        const time = container.querySelector('time') || {};
+        const span = time.parentElement || {};
+        expect(span.textContent).toEqual('27 مهر 1397 - ۱۹ اکتبر ۲۰۱۸');
```
Comment (andrewscfc) [inline packages/containers/psammead-timestamp-container/src/index.test.jsx:100]: These tests are a bit aware of the DOM structure, something React Testing Library goes against a bit, I'd consider  using `expect(container.queryByText('27 مهر 1397 - ۱۹ اکتبر ۲۰۱۸').toBeTruthy();`; this just checks somewhere in the timestamp container, the correct value is present.

(https://testing-library.com/docs/dom-testing-library/api-queries#bytext

**Review by andrewscfc — COMMENTED** [review]
Looks good to me if we can't get a response from apple about the non-standard `text` aria role, it'll fix a practical issue. Ideally we'd follow the standards though so hopefully we'll hear from apple soon: https://github.com/bbc/simorgh/issues/1449#issuecomment-573640710 

Also, minor quibble about the tests, not a big issue though

**Comment by greenc05** [comment]
Lets close this PR, removing semantics of the html with role text seems a step too far, see comment bbc/simorgh#1449

File: packages/containers/psammead-timestamp-container/src/index.test.jsx (hunk 1 above)
Comment (mykeels) [reply packages/containers/psammead-timestamp-container/src/index.test.jsx:100]: The previous text was asserting that the time element's text context had that value. Now, the time element itself has no text content, but its parent does. 

I think container.queryByText(...) opens a loophole when another element is present in the container, with the same text content as we're asserting.

**Pull request closed without merging**